src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -769,6 +775,22 @@ private function registerRequestConfiguration(array $config, ContainerBuilder $c
         }
     }
 
+    private function registerEncryptedSecretsConfiguration(array $config, ContainerBuilder $container)
+    {
+        $container->removeDefinition('secrets.noop_secret_var_processor');
+        $secretsFile = $config['encrypted_secrets']['secrets_file'];
+        $publicKeyFile = $config['encrypted_secrets']['public_key_file'];
+        $privateKeyFile = $config['encrypted_secrets']['private_key_file'];
+        $container->getDefinition('secrets.jwe_handler')
+                  ->addArgument($secretsFile)
+                  ->addArgument($publicKeyFile)
+                  ->addArgument($privateKeyFile);
+
+        $handler = new Reference('secrets.jwe_handler');
+        $container->getDefinition('secrets.secret_var_processor')
+                  ->addArgument($handler);
```

Comment:
This reference/argument should be set in `secrets.xml` - no reason to set it here (it's ok that the actual `secrets.jwe_handler` isn't fully "configured" in `secrets.xml`

Reply:
updated!